Jenkinsfile_parameterized

```diff
@@ -0,0 +1,53 @@
+#!groovy
+@Library("Infrastructure") _
+
+properties([
+    parameters([
+        string(name: 'PRODUCT_NAME', defaultValue: 'ia', description: ''),
+        string(name: 'APP', defaultValue: 'timed-event-service', description: 'Timed Event Service'),
+        string(name: 'TYPE', defaultValue: 'java', description: ''),
+        string(name: 'ENVIRONMENT', defaultValue: 'sandbox', description: 'Environment where code should be build and deployed'),
+        choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in')
+    ]),
+    [$class: 'GithubProjectProperty', projectUrlStr: 'https://github.com/hmcts/timed-event-service.git'],
+    pipelineTriggers([[$class: 'GitHubPushTrigger']])
+])
+
+def secrets = [
+    'ia-${env}': [
+
+        secret('idam-client-id', 'IA_IDAM_CLIENT_ID'),
+        secret('idam-secret', 'IA_IDAM_SECRET'),
+        secret('idam-redirect-uri', 'IA_IDAM_REDIRECT_URI'),
+        secret('s2s-secret', 'IA_S2S_SECRET'),
+        secret('s2s-microservice', 'IA_S2S_MICROSERVICE'),
+        secret('ccd-url', 'CCD_URL'),
+        secret('ccd-gw-url', 'CCD_GW_URL'),
+        secret('dm-url', 'DM_URL'),
+        secret('idam-url', 'IDAM_URL'),
+        secret('s2s-url', 'S2S_URL')
+
+    ]
+]
+
+static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
+    [$class     : 'AzureKeyVaultSecret',
+     secretType : 'Secret',
+     name       : secretName,
+     version    : '',
+     envVariable: envVar
+    ]
+}
+
+withParameterizedPipeline(params.TYPE, params.PRODUCT_NAME, params.APP, params.ENVIRONMENT, params.SUBSCRIPTION) {
+
+    loadVaultSecrets(secrets)
+
+    after('functionalTest:preview') {
```

Comment:
We can discuss later if we need Serenity functional tests for `timed-event-service`